cumulusci/tasks/salesforce/users/permsets.py

```diff
@@ -174,12 +174,34 @@ class AssignPermissionSetLicenses(AssignPermissionSets):
     }
 
     permission_name = "PermissionSetLicense"
-    permission_name_field = "DeveloperName"
+    permission_name_field = ["DeveloperName", "PermissionSetLicenseKey"]
     permission_label = "Permission Set License"
     assignment_name = "PermissionSetLicenseAssign"
     assignment_lookup = "PermissionSetLicenseId"
     assignment_child_relationship = "PermissionSetLicenseAssignments"
 
+    def _get_perm_ids(self):
+        perms_by_ids = {}
+        api_names = "', '".join(self.options["api_names"])
+        missing_perms = {api_name: True for api_name in self.options["api_names"]}
+        for permission_field in self.permission_name_field:
+            perms = self.sf.query(
+                f"SELECT Id,{permission_field} FROM {self.permission_name} WHERE {permission_field} IN ('{api_names}')"
+            )
+            perms_by_ids_subset = {
+                p["Id"]: p[permission_field] for p in perms["records"]
```

Comment:
This function ultimately queries both `DeveloperName` and `PermissionSetLicenseKey`, but it ends up potentially clobbering the `DeveloperName` with `PermissionSetLicenseKey`. That's inconsistent, but probably fine if displaying the actual found name is important for debugging.